core/src/consensus/tendermint/mod.rs

```diff
@@ -319,6 +319,7 @@ impl Tendermint {
                     if let Some(header) = c.block_header(&BlockId::Hash(*hash)) {
                         *self.proposal.write() = Some(*hash);
                         *self.proposal_parent.write() = header.parent_hash();
+                        self.to_step(Step::Prevote);
```

Comment:
Why do we call it here?
We already change the current step in `on_imported_proposal` and `proposal_generated`.

Reply:
It's also my question. There's a code `t.votes.vote(message, author_from_signature);` in `on_message()`. But I'm not sure `to_step(Step::Propose)` called at that time. I cannot find any relations.

Reply:
Since the proposal block could be received in the previous height's prevote or precommit step, we need the `to_step` here.